File: HMW2/univariated_ distribution/univariated_ distribution/Form1.vb
Imports Microsoft.VisualBasic.FileIO

Public Class Form1

    Private array1 As Integer() = {20, 21, 22, 23, 24, 25, 26, 27, 28, 29}
    Private array2 As Integer() = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
    Private Sub RichTextBox1_TextChanged(sender As Object, e As EventArgs) Handles RichTextBox1.TextChanged

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.RichTextBox1.AppendText("Calculating the univariate distribution of the students' age: " & vbLf)
        Me.RichTextBox1.ScrollToCaret()

        Using parser As TextFieldParser = New TextFieldParser("C:\Users\luca1\Desktop\datasetwhireshark.csv")
            parser.TextFieldType = FieldType.Delimited
            parser.SetDelimiters(",")
            Dim header As String() = parser.ReadFields()
            Dim index As Integer = 0

            For i As Integer = 0 To header.Length - 1

                If header(i) = "Count" Then
                    index = i
                End If
            Next

            While Not parser.EndOfData
                Dim fields As String() = parser.ReadFields()

                For i As Double = 0 To array1.Length - 1

                    If array1(i) = Int32.Parse(fields(index)) Then
                        array2(i) += 1
                    End If
                Next
            End While
        End Using

        For i As Integer = 0 To array1.Length - 1
            Dim eta As Integer = array1(i)
            Dim numero As Integer = array2(i)
            Me.RichTextBox1.AppendText(eta.ToString() & ": " & numero.ToString() & vbLf)
        Next
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
End Class

File: HMW3/Homework 3/Form1.cs
using Microsoft.VisualBasic.FileIO; 
using System;
using System.Collections.Generic;

namespace Homework_3
{
    public partial class Form1 : Form
    {
        
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            Dictionary<int, int> univariateData = new Dictionary<int, int>();
            int index = 4;
            this.richTextBox1.AppendText("Lets calculate the univariate distribution of the packets sent: \n");
            this.richTextBox1.ScrollToCaret();
            using (TextFieldParser parser = new TextFieldParser("C:/Users/luca1/Desktop/wiresharkdtasetTCP.csv"))
            {
                parser.TextFieldType = FieldType.Delimited;
                parser.SetDelimiters(",");
                while (!parser.EndOfData)
                {
                    string[] fields = parser.ReadFields();
                    int key = Int32.Parse(fields[index]);
                    if (univariateData.ContainsKey(key))
                    {
                        univariateData[key]++;
                    }
                    else
                    {
                       univariateData.Add(key, 1);
                    }
                }
            }
            foreach (KeyValuePair<int, int> el in univariateData)
            {
                int pacchetti = el.Value;
                int key = el.Key;
                this.richTextBox1.AppendText(key.ToString() + ": " + pacchetti.ToString() + "\n");
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            Dictionary<int, int> univariateData = new Dictionary<int, int>();
            int index = 5;
            this.richTextBox1.AppendText("Lets calculate the univariate distribution of the bytes sent A: \n");
            this.richTextBox1.ScrollToCaret();
            using (TextFieldParser parser = new TextFieldParser("C:/Users/luca1/Desktop/wiresharkdtasetTCP.csv"))
            {
                parser.TextFieldType = FieldType.Delimited;
                parser.SetDelimiters(",");
                while (!parser.EndOfData)
                {
                    string[] fields = parser.ReadFields();
                    int key = Int32.Parse(fields[index]);
                    if (univariateData.ContainsKey(key))
                    {
                        univariateData[key]++;
                    }
                    else
                    {
                        univariateData.Add(key, 1);
                    }
                }
            }
            foreach (KeyValuePair<int, int> el in univariateData)
            {
                int pacchetti = el.Value;
                int key = el.Key;
                this.richTextBox1.AppendText(key.ToString() + ": " + pacchetti.ToString() + "\n");
            }
        }

        private void button3_Click(object sender, EventArgs e)
        {
            this.richTextBox1.Clear();
        }
    }
}